pkg/metrics/otel.go

```diff
@@ -159,6 +159,8 @@ processors:
         action: delete
       - key: configsync.sync.namespace
         action: delete
+      - key: configsync.sync.generation
```

Comment:
This only removes the `generation` tag from the Monarch pipeline. The e2e test for otel-collector is still in progress, currently you need to build and install Config Sync, let the metrics flow and check if deleting it would cause duplicate time series error in the batch.

Reply:
I also suggest adding an attribute processor for GCM pipeline, to temporarily remove this tag from GCM until we have better assessment of the impact. 

Some of the customers are having pretty frequent commit pushes that brings up the cardinality (on record ~300 commits within 30 minutes). IIUC the `generation` is not overlapping with label `commit`, including it in GCM will bump it even higher.

Reply:
I'm waiting for you to merge https://github.com/GoogleContainerTools/kpt-config-sync/pull/748 before I add an aggregation to filter `generation`. The delete is consistent with the previous way of doing it. We can add aggregation as a follow up PR after feature freeze, but before code freeze, or in 1./16.1 if your PR is delayed further.

Yes, generation and commit are non-overlapping, but sync_generation fully overlaps with pod_name. For every pod_name there will be only one sync_generation value. In fact, pod_name is higher cardinality than sync_generation, because the pod can be replaced even when the sync generation has not changed. So adding sync_generation should not increase the cardinality, even if you aggregate out type and commit.

Reply:
It occurred to me that these are custom resource attributes, not normal metrics labels. They're being added with OTEL_RESOURCE_ATTRIBUTES on the otel-agent. I'm pretty sure the provider for both monarch and cloud monitoring will just ignore them, because they're not registered with the server and not hard-coded into the provider.

So they will only show up in prometheus, because we have `resource_to_telemetry_conversion: { enabled: true }`, which converts all resource attributes into labels.

Reply:
I've removed the config in the otel-agent to inject the generation as a metric label. So now it's only being added as a resource attribute, which are ignored by Monarch and Cloud Monitoring otel providers. So the generation should only show up as a label in Prometheus.

Reply:
Not adding this tag in otel-agent attribute processor resolves the concern, confirmed label is not carried into GCM or Monarch.